code/game/objects/items/weapons/medigun.dm

```diff
@@ -0,0 +1,386 @@
+/////Medi-Gun, coded by Deity Link, inspired by Valve's Team Fortress 2/////
+
+#define MAX_UBERCHARGE	100
+#define MEDIHEAL		0.5
+#define MEDIUBER		1
+#define MEDIHARM		-1
+
+/obj/item/medigunpack
+	name = "\improper Medi-Gun"
+	desc = "You wear this on your back and heal people with it."
+	icon = 'icons/obj/gun_experimental.dmi'
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/backpacks_n_bags.dmi', "right_hand" = 'icons/mob/in-hand/right/backpacks_n_bags.dmi')
+	icon_state = "medigunfull"
+	item_state = "medigun"
+	w_class = W_CLASS_LARGE
+	flags = FPRINT
+	slot_flags = SLOT_BACK
+	var/obj/item/medigun/medigun = null
+	var/emagged = 0//emagged gun slowly deals damage of all types at once.
+
+/obj/item/medigunpack/examine(mob/user)
+	..()
+	if(medigun)
+		to_chat(user,"<span class='info'>Charge = [medigun.ubercharge]%</span>")
+
+/obj/item/medigunpack/New()
+	..()
+	medigun = new(src,src)
+
+/obj/item/medigunpack/Destroy()
+	if(medigun)
+		medigun.medigunpack = null
+		qdel(medigun)
+	medigun = null
+	..()
+
+/obj/item/medigunpack/MouseDrop(atom/over_object)
+	if(ishuman(usr))
+		var/mob/living/carbon/human/H = usr
+		if(!H.incapacitated())
+			if (!istype(over_object, /obj/screen/inventory))
+				return ..()
+			playsound(get_turf(src), "rustle", 50, 1, -5)
+
+			if(medigun)
+				if(istype(medigun.loc,/mob))
+					var/mob/M = medigun.loc
+					M.u_equip(medigun)
+				medigun.forceMove(src)
+				update_icon()
+
+			if (src == H.get_item_by_slot(slot_back))
+				var/obj/screen/inventory/OI = over_object
+
+				if(OI.hand_index)
+					H.u_equip(src, 1)
+					H.put_in_hand(OI.hand_index, src)
+					H.update_inv_wear_suit()
+					add_fingerprint(H)
+
+/obj/item/medigunpack/attack_hand(mob/user)
+	if (medigun && (src.loc == user))
+		add_fingerprint(user)
+		user.put_in_hands(medigun)
+		update_icon()
+	else
+		..()
+
+/obj/item/medigunpack/attackby(var/obj/item/weapon/W, var/mob/user)
+	if(W == medigun)
+		user.drop_item(W,src)
+		W.forceMove(W)
+		update_icon()
+		return 1
+	return ..()
+
+/obj/item/medigunpack/emag_act(mob/user as mob)
+	medigun.emag_act(user)
+
+/obj/item/medigunpack/dropped(mob/user)
+	if(medigun)
+		if(istype(medigun.loc,/mob))
+			var/mob/M = medigun.loc
+			M.u_equip(medigun)
+		medigun.forceMove(src)
+		update_icon()
+	else
+		medigun = new(src,src)
+		medigun.emagged = emagged
+
+/obj/item/medigunpack/stripped(mob/wearer,mob/stripper)
+	dropped(wearer)
+
+/obj/item/medigunpack/update_icon()
+	if(medigun && (medigun.loc == src))
+		icon_state = "medigunfull"
+	else
+		icon_state = "medigunpack"
+	if(emagged)
+		icon_state += "red"
+		item_state = "medigunred"
+
+/obj/item/uberdevice
```

Comment:
Because once Skullyton finishes his augment PR we're gonna regroup all our "organ augment" items under a single file, and I might also change the type then. Until then, I can find it back easily in there.